piece-retriever/lib/request.js

```diff
@@ -31,5 +33,36 @@ export function parseRequest(request, { DNS_ROOT }) {
     `Invalid CID: ${pieceCid}. It is not a valid CommP (v1 or v2).`,
   )
 
-  return { payerWalletAddress, pieceCid }
+  const botName = checkBotAuthorization(request, { BOT_TOKENS })
+
+  return { payerWalletAddress, pieceCid, botName }
+}
+
+/**
+ * @param {Request} request
+ * @param {object} args
+ * @param {string} args.BOT_TOKENS
+ * @returns {string | undefined} Bot name or the access token
+ */
+export function checkBotAuthorization(request, { BOT_TOKENS }) {
+  const allowedTokens = BOT_TOKENS.split(',').map((t) => t.trim())
```

Comment:
I agree with @juliangruber here, using JSON here instead of custom string format would be nicer